src/doc/trpl/guessing-game.md

```diff
@@ -0,0 +1,991 @@
+
+# Processing a Guess
+
+Let’s get to it! The first thing we need to do for our guessing game is
+allow our player to input a guess. Put this in your `src/main.rs`:
+
+```rust,no_run
+use std::io;
+
+fn main() {
+    println!("Guess the number!");
+
+    println!("Please input your guess.");
+
+    let mut guess = String::new();
+
+    let input = io::stdin().read_line(&mut guess)
+        .ok()
+        .expect("Failed to read line");
+
+    println!("You guessed: {}", input);
+}
+```
+
+There’s a lot here! Let’s go over it, bit by bit.
+
+```rust,ignore
+use std::io;
+```
+
+We’ll need to take user input, and then print the result as output. As such, we
+need the `io` library from the standard library. Rust only imports a few things
+into every program, [the ‘prelude’][prelude]. If it’s not in the prelude,
+you’ll have to `use` it directly.
+
+[prelude]: ../std/prelude/index.html
+
+```rust,ignore
+fn main() {
+```
+
+As you’ve seen before, the `main()` function is the entry point into your
+program. The `fn` syntax declares a new function, the `()`s indicate that
+there are no arguments, and `{` starts the body of the function. Because
+we didn’t include a return type, it’s assumed to be `()`, an empty
+[tuple][tuples].
+
+[tuples]: primitive-types.html#tuples
+
+```rust,ignore
+    println!("Guess the number!");
+
+    println!("Please input your guess.");
+```
+
+We previously learned that `println!()` is a [macro][macros] that
+prints a [string][strings] to the screen.
+
+[macros]: macros.html
+[strings]: strings.html
+
+```rust,ignore
+    let mut guess = String::new();
+```
+
+Now we’re getting interesting! There’s a lot going on in this little line. The first thing to notice is that this is a [let statement][let], which is used to create ‘variable bindings’. They take this form:
+
+```rust,ignore
+let foo = bar;
+```
+
+[let]: variable-bindings.html
+
+This will create a new binding named `foo`, and bind it to the value `bar`. In
+many languages, this is called a ‘variable’, but Rust’s variable bindings have
+a few tricks up their sleeves.
+
+For example, they’re [immutable][immutable] by default. That’s why our example
+uses `mut`: it makes a binding mutable, rather than immutable. `let` doesn’t
+take a name on the left hand side, it actually accepts a
+‘[pattern][patterns]’. We’ll use patterns more later. It’s easy enough
+to use for now:
+
+```
+let foo = 5; // immutable.
+let mut bar = 5; // mutable
+```
+
+[immutable]: mutability.html
+[patterns]: patterns.html
+
+Oh, and `//` will start a comment, until the end of the line. Rust ignores
+everything in [comments][comments].
+
+[comments]: comments.html
+
+So now we know that `let mut guess` will introduce a mutable binding named
+`guess`, but we have to look at the other side of the `=` for what it’s
+bound to: `String::new()`.
+
+`String` is a string type, provided by the standard library. A
+[`String`][string] is a growable, UTF-8 encoded string.
```

Comment:
> `String` is a string type

doesn't really convey much meaning unless you already know what a string is. Maybe add "string of characters" or "text" in there somewhere? (I'm not sure what the audience of this text is, maybe you don't need to explain strings…)